catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java

```diff
@@ -331,50 +312,92 @@ private void commitUpdates(StorageRequest request, String id, Path requestIdDir)
                             "Unable to move files by simple rename, resorting to copy. This will impact performance.",
                             e);
                     try {
-                        if (Files.deleteIfExists(target)) {
-                            LOGGER.debug("Remove existing content id directory for commit: " + id);
-                        }
                         Path createdTarget = Files.createDirectories(target);
                         List<Path> files = listPaths(contentIdDir);
-                        Files.copy(files.get(0), Paths.get(createdTarget.toAbsolutePath()
-                                .toString(), files.get(0)
-                                .getFileName()
-                                .toString()));
+                        Files.copy(files.get(0),
+                                Paths.get(createdTarget.toAbsolutePath()
+                                                .toString(),
+                                        files.get(0)
+                                                .getFileName()
+                                                .toString()));
                     } catch (IOException e1) {
-                        throw new StorageException("Unable to commit changes for request: " + id,
-                                e1);
+                        throw new StorageException(
+                                "Unable to commit changes for request: " + request.getId(), e1);
                     }
                 }
             }
+        } catch (URISyntaxException e) {
+            throw new StorageException(e);
         } finally {
-            rollback(request);
+            clearStage(request);
         }
     }
 
     @Override
-    public synchronized void rollback(StorageRequest request) throws StorageException {
+    public synchronized void clearStage(StorageRequest request) throws StorageException {
```

Comment:
Change this back to rollback. That was the name that several of us decided on in the previous PR.